### DISCUSSION TITLE
fortios-firewall-stats3 // 404 on FortiOS v7

### DISCUSSION THREAD

**Discussion by ne0xit** [Linuxfabrik/monitoring-plugins#614] [Q&A]
we have just updated our forti to forti 7.0.6. 

Our Icinga check now throws a 404 for the firewall stats
HTTP error "404 Not Found" while fetching https://123.123.123.123/api/v2/monitor/firewall/policy6/select/?access_token********

How can i fix this error?

Tag: 2022072001

**Comment by markuslf** [comment]
Let me ask two more counter questions:
* Which version was running on the Forti before the upgrade?
* Did you (or the upgrade) disable IPv6?

**Comment by ne0xit** [comment]
Hi @markuslf,

thank you for your answer. 
- The previously installed version was 6.2.10. The fortigate upgrade path was followed. 
- IPv6 is still activated and also works

**Comment by markuslf** [comment]
So it could be that the API URL to query the statistics data has changed - and possibly some other URLs.
Currently our FortiOS checks query these URLs:
* `/api/v2/cmdb/system/global`
* `/api/v2/monitor/firewall/policy/select/`
* `/api/v2/monitor/firewall/policy6/select/`
* `/api/v2/monitor/system/firmware/`
* `/api/v2/monitor/system/ha-statistics/select/`
* `/api/v2/monitor/system/interface/select`
* `/api/v2/monitor/system/resource/usage?resource=cpu&interval=1-min`
* `/api/v2/monitor/system/resource/usage?resource=mem&interval=1-min`
* `/api/v2/monitor/system/sensor-info/select`

Now it would be nice to know what the new API URLs are (if they have changed). Do you have new Forti documentation available?

Since we don't have Forti in use ourselves, we can only offer to fix this remotely on a time and effort basis. Of course, the improved checks will then be published here on GitHub as usual.
